src/components/shared/LinkButton.tsx

```diff
@@ -0,0 +1,105 @@
+import Link from "@docusaurus/Link"
+import {Theme} from "@site/src/constants"
+import React from "react"
+import {SVGProps} from "react"
+
+type LinkButtonProps = {
+  title?: string
+  Icon?: React.ComponentType<SVGProps<SVGSVGElement>> // Define the type of Icon prop
+  theme: Theme
+  onClick?: () => void | Promise<void>
+  href?: string
+  width?: "small" | "medium" | "large" | "auto"
+  disabled?: boolean
+}
+
+const LinkButton = ({title, Icon, theme, onClick, href, width = "auto", disabled}: LinkButtonProps): JSX.Element => {
+  // Generate button widths as tailwind is not able to handle dynamic widths
+  const setButtonWidth = () => {
+    switch (width) {
+      case "small":
+        return "w-[228px]"
+      case "medium":
+        return "w-[300px]"
+      case "large":
+        return "w-[500px]"
+      case "auto":
+        return "w-fit"
+      default:
+        return "w-fit"
+    }
+  }
+
+  // Generate classes based on the provided theme
+  const generateThemeClasses = () => {
+    const themes = {
+      [Theme.Light]: {
+        classes:
+          "border border-solid border-tailCall-border-dark-100 text-tailCall-dark-500 bg-transparent hover:text-tailCall-dark-500",
+        gridClasses: "",
+      },
+      [Theme.Dark]: {
+        classes:
+          "border-2 border-solid border-tailCall-border-dark-100 text-tailCall-light-100 bg-white hover:text-tailCall-light-100",
+        gridClasses: "",
+      },
+      [Theme.Gray]: {
+        classes:
+          "border-2 border-solid border-tailCall-light-100 text-tailCall-light-100 bg-transparent hover:text-tailCall-light-100",
+        gridClasses: "hidden",
+      },
+    }
+
+    return themes[theme] || {classes: "", styles: "", gridClasses: ""}
+  }
+
+  const renderBackgroundElements = () => {
```

Comment:
simplify this function for redability

Reply:
pass theme to this function make it pure

Reply:
added if else statement and made the fn pure